<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Use case: LC-MS Based Approaches to Investigate Metabolomic Differences in the Urine of Young Women after Drinking Cranberry Juice or Apple Juice • fobitools</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/spacelab/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Use case: LC-MS Based Approaches to Investigate Metabolomic Differences in the Urine of Young Women after Drinking Cranberry Juice or Apple Juice">
<meta property="og:description" content="fobitools">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">fobitools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.11.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/food_enrichment_analysis.html">Simple food ORA
</a>
    </li>
    <li>
      <a href="../articles/MW_ST000291_enrichment.html">Use case ST000291
</a>
    </li>
    <li>
      <a href="../articles/MW_ST000629_enrichment.html">Use case ST000629
</a>
    </li>
    <li>
      <a href="../articles/dietary_data_annotation.html">Dietary text annotation
</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/pcastellanoescuder/fobitools" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="MW_ST000291_enrichment_files/kePrint-0.0.1/kePrint.js"></script><link href="MW_ST000291_enrichment_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Use case: LC-MS Based Approaches to Investigate
Metabolomic Differences in the Urine of Young Women after Drinking
Cranberry Juice or Apple Juice</h1>
                        <h4 data-toc-skip class="author">Pol
Castellano-Escuder</h4>
            <address class="author_afil">
      Duke
University<br><a class="author_email" href="mailto:#"></a><a href="mailto:polcaes@gmail.com" class="email">polcaes@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">6 May 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pcastellanoescuder/fobitools/blob/HEAD/vignettes/MW_ST000291_enrichment.Rmd" class="external-link"><code>vignettes/MW_ST000291_enrichment.Rmd</code></a></small>
      <div class="hidden name"><code>MW_ST000291_enrichment.Rmd</code></div>

    </div>

    
    
<p><strong>Compiled date</strong>: 2024-05-06</p>
<p><strong>Last edited</strong>: 2024-02-12</p>
<p><strong>License</strong>: GPL-3</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Run the following code to install the Bioconductor version of the
package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"fobitools"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-packages">Load packages<a class="anchor" aria-label="anchor" href="#load-packages"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/pcastellanoescuder/fobitools/" class="external-link">fobitools</a></span><span class="op">)</span></span></code></pre></div>
<p>We will also need some additional <a href="https://cran.r-project.org" class="external-link">CRAN</a> and <a href="https://bioconductor.org" class="external-link">Bioconductor</a> packages for performing
tasks such as statistical analysis and web scraping.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># CRAN</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rvest.tidyverse.org/" class="external-link">rvest</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggrepel.slowkow.com/" class="external-link">ggrepel</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://haozhu233.github.io/kableExtra/" class="external-link">kableExtra</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Bioconductor</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/pcastellanoescuder/POMA" class="external-link">POMA</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">metabolomicsWorkbenchR</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="download-the-data-from-metabolomics-workbench">Download the data from Metabolomics Workbench<a class="anchor" aria-label="anchor" href="#download-the-data-from-metabolomics-workbench"></a>
</h2>
<p>The Metabolomics Workbench, available at <a href="www.metabolomicsworkbench.org">www.metabolomicsworkbench.org</a>,
is a public repository for metabolomics metadata and experimental data
spanning various species and experimental platforms, metabolite
standards, metabolite structures, protocols, tutorials, and training
material and other educational resources. It provides a computational
platform to integrate, analyze, track, deposit and disseminate large
volumes of heterogeneous data from a wide variety of metabolomics
studies including mass spectrometry (MS) and nuclear magnetic resonance
spectrometry (NMR) data spanning over 20 different species covering all
the major taxonomic categories including humans and other mammals,
plants, insects, invertebrates and microorganisms <span class="citation">(<a href="#ref-sud2016metabolomics">Sud et al.
2016</a>)</span>.</p>
<p>The <code>metabolomicsWorkbenchR</code> Bioconductor package allows
us to obtain data from the Metabolomics Workbench repository. In this
vignette we will use the sample data set <a href="https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Study&amp;StudyID=ST000291" class="external-link">ST000291</a>.</p>
<div class="section level3">
<h3 id="sumary-of-the-study-st000291">Sumary of the study (ST000291)<a class="anchor" aria-label="anchor" href="#sumary-of-the-study-st000291"></a>
</h3>
<p>Eighteen healthy female college students between 21-29 years old with
a normal BMI of 18.5-25 were recruited. Each subject was provided with a
list of foods that contained significant amount of procyanidins, such as
<strong>cranberries, apples, grapes, blueberries, chocolate and
plums</strong>. They were advised to avoid these foods during the 1-6th
day and the rest of the study. On the morning of the 7th day, a
first-morning baseline urine sample and blood sample were collected from
all human subjects after overnight fasting. <strong>Participants were
then randomly allocated into two groups (n=9) to consume cranberry juice
or apple juice</strong>. Six bottles (250 ml/bottle) of juice were given
to participants to drink in the morning and evening of the 7th, 8th, and
9th day. On the morning of 10th day, all subjects returned to the
clinical unit to provide a first-morning urine sample after overnight
fasting. The blood sample was also collected from participants 30 min
later after they drank another bottle of juice in the morning. After
two-weeks of wash out period, participants switched to the alternative
regimen and repeated the protocol. One human subject was dropped off
this study because she missed part of her appointments. Another two
human subjects were removed from urine metabolomics analyses because
they failed to provide required urine samples after juice drinking.The
present study aimed to investigate overall metabolic changes caused by
procyanidins concentrates from cranberries and apples using a global
LCMS based metabolomics approach. All plasma and urine samples were
stored at -80ºC until analysis.</p>
</div>
<div class="section level3">
<h3 id="download-data">Download data<a class="anchor" aria-label="anchor" href="#download-data"></a>
</h3>
<p>This study is composed of two complementary MS analyses, the positive
mode (AN000464) and the negative mode (AN000465). Let’s download them
both!</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_negative_mode</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/metabolomicsWorkbenchR/man/do_query.html" class="external-link">do_query</a></span><span class="op">(</span></span>
<span>  context <span class="op">=</span> <span class="st">"study"</span>,</span>
<span>  input_item <span class="op">=</span> <span class="st">"analysis_id"</span>,</span>
<span>  input_value <span class="op">=</span> <span class="st">"AN000465"</span>,</span>
<span>  output_item <span class="op">=</span> <span class="st">"SummarizedExperiment"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">data_positive_mode</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/metabolomicsWorkbenchR/man/do_query.html" class="external-link">do_query</a></span><span class="op">(</span></span>
<span>  context <span class="op">=</span> <span class="st">"study"</span>,</span>
<span>  input_item <span class="op">=</span> <span class="st">"analysis_id"</span>,</span>
<span>  input_value <span class="op">=</span> <span class="st">"AN000464"</span>,</span>
<span>  output_item <span class="op">=</span> <span class="st">"SummarizedExperiment"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="scraping-metabolite-names-and-identifiers-with-rvest">Scraping metabolite names and identifiers with
<code>rvest</code><a class="anchor" aria-label="anchor" href="#scraping-metabolite-names-and-identifiers-with-rvest"></a>
</h3>
<p>In many metabolomics studies, the reproducibility of analyses is
severely affected by the poor interoperability of metabolite names and
their identifiers. For this reason it is important to develop tools that
facilitate the process of converting one type of identifier to another.
In order to use the <code>fobitools</code> package, we need some generic
identifier (such as PubChem, KEGG or HMDB) that allows us to obtain the
corresponding FOBI identifier for each metabolite. The Metabolomics
Workbench repository provides us with this information for many of the
metabolites quantified in study <a href="https://www.metabolomicsworkbench.org/data/show_metabolites_by_study.php?STUDY_ID=ST000291&amp;SEARCH_TYPE=KNOWN&amp;STUDY_TYPE=MS&amp;RESULT_TYPE=1" class="external-link">ST000291</a>
(Figure @ref(fig:metabolitenames)). In order to easily obtain this
information, we will perform a web scraping operation using the
<code>rvest</code> package.</p>
<div class="figure" style="text-align: center">
<img src="figure/ST000291_names.png" alt="Metabolite identifiers of the ST000291 Metabolomics Workbench study." width="100%"><p class="caption">
Metabolite identifiers of the ST000291 Metabolomics Workbench study.
</p>
</div>
<p>Below we obtain the PubChem and KEGG identifiers of the metabolites
analyzed in the positive and negative mode directly from the
Metabolomics Workbench website. We will then remove those duplicate
identifiers.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metaboliteNamesURL</span> <span class="op">&lt;-</span> <span class="st">"https://www.metabolomicsworkbench.org/data/show_metabolites_by_study.php?STUDY_ID=ST000291&amp;SEARCH_TYPE=KNOWN&amp;STUDY_TYPE=MS&amp;RESULT_TYPE=1"</span></span>
<span><span class="va">metaboliteNames</span> <span class="op">&lt;-</span> <span class="va">metaboliteNamesURL</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="http://xml2.r-lib.org/reference/read_xml.html" class="external-link">read_html</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rvest.tidyverse.org/reference/rename.html" class="external-link">html_nodes</a></span><span class="op">(</span><span class="st">".datatable"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">metaboliteNames_negative</span> <span class="op">&lt;-</span> <span class="va">metaboliteNames</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rvest.tidyverse.org/reference/html_table.html" class="external-link">html_table</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">`Metabolite Name`</span>, <span class="va">PubChemCompound_ID</span>, <span class="va">`Kegg Id`</span><span class="op">)</span></span>
<span></span>
<span><span class="va">metaboliteNames_positive</span> <span class="op">&lt;-</span> <span class="va">metaboliteNames</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="va">.</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rvest.tidyverse.org/reference/html_table.html" class="external-link">html_table</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">`Metabolite Name`</span>, <span class="va">PubChemCompound_ID</span>, <span class="va">`Kegg Id`</span><span class="op">)</span></span>
<span></span>
<span><span class="va">metaboliteNames</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind_rows.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="va">metaboliteNames_negative</span>, <span class="va">metaboliteNames_positive</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>names <span class="op">=</span> <span class="fl">1</span>, PubChem <span class="op">=</span> <span class="fl">2</span>, KEGG <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>KEGG <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">KEGG</span> <span class="op">==</span> <span class="st">"-"</span>, <span class="st">"UNKNOWN"</span>, <span class="va">KEGG</span><span class="op">)</span>,</span>
<span>         PubChem <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">PubChem</span> <span class="op">==</span> <span class="st">"-"</span>, <span class="st">"UNKNOWN"</span>, <span class="va">PubChem</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/duplicated.html" class="external-link">duplicated</a></span><span class="op">(</span><span class="va">PubChem</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="prepare-features-and-metadata">Prepare features and metadata<a class="anchor" aria-label="anchor" href="#prepare-features-and-metadata"></a>
</h2>
<p>Now we have to prepare the metadata and features in order to proceed
with the statistical analysis. In this step we assign to each metabolite
its PubChem identifier obtained in the previous step.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## negative mode features </span></span>
<span><span class="va">features_negative</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">data_negative_mode</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">features_negative</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">data_negative_mode</span><span class="op">)</span><span class="op">$</span><span class="va">metabolite</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">data_negative_mode</span><span class="op">)</span><span class="op">$</span><span class="va">metabolite</span><span class="op">)</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## positive mode features</span></span>
<span><span class="va">features_positive</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">data_positive_mode</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">features_positive</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">data_positive_mode</span><span class="op">)</span><span class="op">$</span><span class="va">metabolite</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">data_positive_mode</span><span class="op">)</span><span class="op">$</span><span class="va">metabolite</span><span class="op">)</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co">## combine positive and negative mode and set PubChem IDs as feature names</span></span>
<span><span class="va">features</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind_rows.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="va">features_negative</span>, <span class="va">features_positive</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span><span class="st">"names"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">right_join</a></span><span class="op">(</span><span class="va">metaboliteNames</span>, by <span class="op">=</span> <span class="st">"names"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">names</span>, <span class="op">-</span><span class="va">KEGG</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"PubChem"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## metadata</span></span>
<span><span class="va">pdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">data_negative_mode</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co"># or "data_positive_mode". They are equal</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span><span class="st">"ID"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Treatment <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html" class="external-link">case_when</a></span><span class="op">(</span><span class="va">Treatment</span> <span class="op">==</span> <span class="st">"Baseline urine"</span> <span class="op">~</span> <span class="st">"Baseline"</span>,</span>
<span>                               <span class="va">Treatment</span> <span class="op">==</span> <span class="st">"Urine after drinking apple juice"</span> <span class="op">~</span> <span class="st">"Apple"</span>,</span>
<span>                               <span class="va">Treatment</span> <span class="op">==</span> <span class="st">"Urine after drinking cranberry juice"</span> <span class="op">~</span> <span class="st">"Cranberry"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="statistical-analysis-with-poma">Statistical analysis with <code>POMA</code><a class="anchor" aria-label="anchor" href="#statistical-analysis-with-poma"></a>
</h2>
<p><code>POMA</code> provides a structured, reproducible and easy-to-use
workflow for the visualization, preprocessing, exploration, and
statistical analysis of metabolomics and proteomics data. The main aim
of this package is to enable a flexible data cleaning and statistical
analysis processes in one comprehensible and user-friendly R package.
<code>POMA</code> uses the standardized
<code>SummarizedExperiment</code> data structures, to achieve the
maximum flexibility and reproducibility and makes <code>POMA</code>
compatible with other Bioconductor packages <span class="citation">(<a href="#ref-POMA">Castellano-Escuder, Andrés-Lacueva, and Sánchez-Pla
2021</a>)</span>.</p>
<div class="section level3">
<h3 id="create-a-summarizedexperiment-object">Create a <code>SummarizedExperiment</code> object<a class="anchor" aria-label="anchor" href="#create-a-summarizedexperiment-object"></a>
</h3>
<p>First, we create a <code>SummarizedExperiment</code> object that
integrates both metadata and features in the same data structure.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_sumexp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/POMA/man/PomaCreateObject.html" class="external-link">PomaCreateObject</a></span><span class="op">(</span>metadata <span class="op">=</span> <span class="va">pdata</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">features</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="preprocessing">Preprocessing<a class="anchor" aria-label="anchor" href="#preprocessing"></a>
</h3>
<p>Second, we perform the preprocessing step. This step includes the
missing value imputation unsing the <span class="math inline">\(k\)</span>-NN algorithm and log Pareto
normalization (transformation and scaling). Once these steps are
completed, we can proceed to the statistical analysis of these data.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_preprocessed</span> <span class="op">&lt;-</span> <span class="va">data_sumexp</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/POMA/man/PomaImpute.html" class="external-link">PomaImpute</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/POMA/man/PomaNorm.html" class="external-link">PomaNorm</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="limma-model">Limma model<a class="anchor" aria-label="anchor" href="#limma-model"></a>
</h3>
<p>We use a limma model <span class="citation">(<a href="#ref-limma">Ritchie et al. 2015</a>)</span> to identify those most
significant metabolites between the “<strong>Baseline urine</strong>”
and “<strong>Urine after drinking cranberry juice</strong>” groups. With
this analysis we expect to find metabolites related to cranberry
intake.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">limma_res</span> <span class="op">&lt;-</span> <span class="va">data_preprocessed</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/POMA/man/PomaLimma.html" class="external-link">PomaLimma</a></span><span class="op">(</span>contrast <span class="op">=</span> <span class="st">"Baseline-Cranberry"</span>, adjust <span class="op">=</span> <span class="st">"fdr"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>PubChemCID <span class="op">=</span> <span class="va">feature</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>PubChemCID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"X"</span>, <span class="st">""</span>, <span class="va">PubChemCID</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># show the first 10 features</span></span>
<span><span class="va">limma_res</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">1L</span><span class="op">:</span><span class="fl">10L</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kbl.html" class="external-link">kbl</a></span><span class="op">(</span>row.names <span class="op">=</span> <span class="cn">FALSE</span>, booktabs <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html" class="external-link">kable_styling</a></span><span class="op">(</span>latex_options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"striped"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
PubChemCID
</th>
<th style="text-align:right;">
logFC
</th>
<th style="text-align:right;">
AveExpr
</th>
<th style="text-align:right;">
t
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
adj_pvalue
</th>
<th style="text-align:right;">
B
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
54678503
</td>
<td style="text-align:right;">
-1.922260
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
-6.446862
</td>
<td style="text-align:right;">
1.0e-07
</td>
<td style="text-align:right;">
0.0000847
</td>
<td style="text-align:right;">
8.108057
</td>
</tr>
<tr>
<td style="text-align:left;">
71485
</td>
<td style="text-align:right;">
-2.153637
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
-5.853017
</td>
<td style="text-align:right;">
5.0e-07
</td>
<td style="text-align:right;">
0.0003289
</td>
<td style="text-align:right;">
6.194381
</td>
</tr>
<tr>
<td style="text-align:left;">
94214
</td>
<td style="text-align:right;">
-1.436888
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
-5.691638
</td>
<td style="text-align:right;">
8.0e-07
</td>
<td style="text-align:right;">
0.0003817
</td>
<td style="text-align:right;">
5.676529
</td>
</tr>
<tr>
<td style="text-align:left;">
5378303
</td>
<td style="text-align:right;">
-1.591375
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
-5.543843
</td>
<td style="text-align:right;">
1.4e-06
</td>
<td style="text-align:right;">
0.0004030
</td>
<td style="text-align:right;">
5.203836
</td>
</tr>
<tr>
<td style="text-align:left;">
3037
</td>
<td style="text-align:right;">
-2.095435
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
-5.463171
</td>
<td style="text-align:right;">
1.8e-06
</td>
<td style="text-align:right;">
0.0004030
</td>
<td style="text-align:right;">
4.946595
</td>
</tr>
<tr>
<td style="text-align:left;">
10178
</td>
<td style="text-align:right;">
-2.195515
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
-5.454885
</td>
<td style="text-align:right;">
1.9e-06
</td>
<td style="text-align:right;">
0.0004030
</td>
<td style="text-align:right;">
4.920207
</td>
</tr>
<tr>
<td style="text-align:left;">
443071
</td>
<td style="text-align:right;">
-2.382564
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
-5.402912
</td>
<td style="text-align:right;">
2.3e-06
</td>
<td style="text-align:right;">
0.0004030
</td>
<td style="text-align:right;">
4.754851
</td>
</tr>
<tr>
<td style="text-align:left;">
5486800
</td>
<td style="text-align:right;">
1.175814
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
5.388792
</td>
<td style="text-align:right;">
2.4e-06
</td>
<td style="text-align:right;">
0.0004030
</td>
<td style="text-align:right;">
4.709975
</td>
</tr>
<tr>
<td style="text-align:left;">
17531
</td>
<td style="text-align:right;">
-1.490917
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
-5.318449
</td>
<td style="text-align:right;">
3.0e-06
</td>
<td style="text-align:right;">
0.0004550
</td>
<td style="text-align:right;">
4.486739
</td>
</tr>
<tr>
<td style="text-align:left;">
3035199
</td>
<td style="text-align:right;">
-1.298588
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
-5.048492
</td>
<td style="text-align:right;">
7.5e-06
</td>
<td style="text-align:right;">
0.0010196
</td>
<td style="text-align:right;">
3.635733
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="convert-pubchem-ids-to-fobi-ids">Convert PubChem IDs to FOBI IDs<a class="anchor" aria-label="anchor" href="#convert-pubchem-ids-to-fobi-ids"></a>
</h2>
<p>Once we have the results of the statistical analysis and generic
identifiers recognized in the FOBI ontology <span class="citation">(<a href="#ref-castellano2020fobi">Castellano-Escuder et al.
2020</a>)</span>, we can proceed to perform one of the main functions
provided by the <code>fobitools</code> package, the ID conversion. With
the <code><a href="../reference/id_convert.html">fobitools::id_convert()</a></code> command, users can convert
different IDs between FOBI, HMDB, KEGG, PubChem, InChIKey, InChICode,
ChemSpider, and chemical names. We will then obtain the FOBI IDs from
the PubChem IDs (obtained in the previous sections) and add them as a
new column to the results of the limma model.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">limma_FOBI_names</span> <span class="op">&lt;-</span> <span class="va">limma_res</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="st">"PubChemCID"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">fobitools</span><span class="fu">::</span><span class="fu"><a href="../reference/id_convert.html">id_convert</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># show the ID conversion results</span></span>
<span><span class="va">limma_FOBI_names</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kbl.html" class="external-link">kbl</a></span><span class="op">(</span>row.names <span class="op">=</span> <span class="cn">FALSE</span>, booktabs <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html" class="external-link">kable_styling</a></span><span class="op">(</span>latex_options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"striped"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
FOBI
</th>
<th style="text-align:left;">
PubChemCID
</th>
<th style="text-align:left;">
ChemSpider
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
FOBI:030415
</td>
<td style="text-align:left;">
91
</td>
<td style="text-align:left;">
89
</td>
</tr>
<tr>
<td style="text-align:left;">
FOBI:030711
</td>
<td style="text-align:left;">
1145
</td>
<td style="text-align:left;">
1113
</td>
</tr>
<tr>
<td style="text-align:left;">
FOBI:030555
</td>
<td style="text-align:left;">
5280445
</td>
<td style="text-align:left;">
4444102
</td>
</tr>
<tr>
<td style="text-align:left;">
FOBI:030709
</td>
<td style="text-align:left;">
1123
</td>
<td style="text-align:left;">
10675782
</td>
</tr>
<tr>
<td style="text-align:left;">
FOBI:030625
</td>
<td style="text-align:left;">
7533
</td>
<td style="text-align:left;">
15484224
</td>
</tr>
<tr>
<td style="text-align:left;">
FOBI:030397
</td>
<td style="text-align:left;">
1794427
</td>
<td style="text-align:left;">
1405788
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">limma_FOBI_names</span> <span class="op">&lt;-</span> <span class="va">limma_FOBI_names</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">right_join</a></span><span class="op">(</span><span class="va">limma_res</span>, by <span class="op">=</span> <span class="st">"PubChemCID"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="op">-</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">pvalue</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="enrichment-analysis">Enrichment analysis<a class="anchor" aria-label="anchor" href="#enrichment-analysis"></a>
</h2>
<p>Enrichment analysis denotes any method that benefits from biological
pathway or network information to gain insight into a biological system
<span class="citation">(<a href="#ref-creixell2015pathway">Creixell et
al. 2015</a>)</span>. In other words, these type of analyses integrate
the existing biological knowledge (from different biological sources
such as databases and ontologies) and the statistical results of
<em>omics</em> studies, obtaining a deeper understanding of biological
systems.</p>
<p>In most metabolomics studies, the output of statistical analysis is
usually a list of features selected as statistically significant or
statistically relevant according to a pre-defined statistical criteria.
Enrichment analysis methods use these selected features to explore
associated biologically relevant pathways, diseases, etc., depending on
the nature of the input feature list (genes, metabolites, etc.) and the
source used to extract the biological knowledge (GO, KEGG,
<strong>FOBI</strong>, etc.).</p>
<p>Here, we present a tool that uses the FOBI information to perform
different types of enrichment analyses. Therefore, the presented methods
allow researchers to move from lists of metabolites to chemical classes
and food groups associated with those lists, and consequently, to the
study design.</p>
<p>Currently, the most popular used approaches for enrichment analysis
are the over representation analysis (ORA) and the gene set enrichment
analysis (GSEA), with its variants for other fields such as the
metabolite set enrichment analysis (MSEA) <span class="citation">(<a href="#ref-xia2010msea">Xia and Wishart 2010</a>)</span>.</p>
<div class="section level3">
<h3 id="over-representation-analysis-ora">Over representation analysis (ORA)<a class="anchor" aria-label="anchor" href="#over-representation-analysis-ora"></a>
</h3>
<p>ORA is one of the most used methods to perform enrichment analysis in
metabolomics studies due to its simplicity and easy understanding. This
method statistically evaluates the fraction of metabolites in a
particular pathway found among the set of metabolites statistically
selected. Thus, ORA is used to test if certain groups of metabolites are
represented more than expected by chance given a feature list.</p>
<p>However, ORA has a number of limitations. The most important one is
the need of using a certain threshold or criteria to select the feature
list. This means that metabolites do not meet the selection criteria
must be discarded. The second big limitation of ORA is that this method
assumes independence of sets and features. In ORA, is assumed that each
feature is independent of the other features and each set is independent
of the other sets.</p>
<p>Here, we perform an ORA with the <code>fobitools</code> package,
where we will use as a universe all the metabolites of the study present
in FOBI and as a list those metabolites with a raw p-value &lt; 0.01 in
the limma results table.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metaboliteList</span> <span class="op">&lt;-</span> <span class="va">limma_FOBI_names</span><span class="op">$</span><span class="va">FOBI</span><span class="op">[</span><span class="va">limma_FOBI_names</span><span class="op">$</span><span class="va">pvalue</span> <span class="op">&lt;</span> <span class="fl">0.01</span><span class="op">]</span></span>
<span><span class="va">metaboliteUniverse</span> <span class="op">&lt;-</span> <span class="va">limma_FOBI_names</span><span class="op">$</span><span class="va">FOBI</span></span>
<span></span>
<span><span class="fu">fobitools</span><span class="fu">::</span><span class="fu"><a href="../reference/ora.html">ora</a></span><span class="op">(</span>metaboliteList <span class="op">=</span> <span class="va">metaboliteList</span>,</span>
<span>               metaboliteUniverse <span class="op">=</span> <span class="va">metaboliteUniverse</span>,</span>
<span>               pvalCutoff <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kbl.html" class="external-link">kbl</a></span><span class="op">(</span>row.names <span class="op">=</span> <span class="cn">FALSE</span>, booktabs <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html" class="external-link">kable_styling</a></span><span class="op">(</span>latex_options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"striped"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
className
</th>
<th style="text-align:right;">
classSize
</th>
<th style="text-align:right;">
overlap
</th>
<th style="text-align:right;">
pval
</th>
<th style="text-align:right;">
padj
</th>
<th style="text-align:left;">
overlapMetabolites
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
soft drink (dietetic)
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.1362126
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
FOBI:030627
</td>
</tr>
<tr>
<td style="text-align:left;">
cocoa
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.3164249
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
FOBI:050272
</td>
</tr>
<tr>
<td style="text-align:left;">
coffee based beverage product
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
0.3416255
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
FOBI:030….
</td>
</tr>
<tr>
<td style="text-align:left;">
coffee (liquid drink)
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.4214407
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
FOBI:050272
</td>
</tr>
</tbody>
</table>
<p>As we can see, due to the limitations of this methodology and the
small number of metabolites that meet the set statistical criterion, the
results do not show a clear and obvious relationship with the design of
the study, as the food groups that appear in the ORA results do not
correspond to those foods administered in the intervention.</p>
</div>
<div class="section level3">
<h3 id="msea">MSEA<a class="anchor" aria-label="anchor" href="#msea"></a>
</h3>
<p>Gene Set Enrichment Analysis (GSEA) methodology was proposed for the
first time in 2005, with the aim of improving the interpretation of gene
expression data. The main purpose of GSEA is to determine whether
members of a gene set <span class="math inline">\(S\)</span> tend to
occur toward the top (or bottom) of the gene list <span class="math inline">\(L\)</span>, in which case the gene set is
correlated with the phenotypic class distinction <span class="citation">(<a href="#ref-subramanian2005gene">Subramanian et al.
2005</a>)</span>.</p>
<p>This type of analysis basically consists of three key steps <span class="citation">(<a href="#ref-subramanian2005gene">Subramanian et al.
2005</a>)</span>:</p>
<p>The first step consists on the calculation of an enrichment score
(<span class="math inline">\(ES\)</span>). This value indicates the
degree to which a set <span class="math inline">\(S\)</span> is
overrepresented at the extremes (top or bottom) of the entire ranked
gene list <span class="math inline">\(L\)</span>. The <span class="math inline">\(ES\)</span> is calculated by walking down the list
<span class="math inline">\(L\)</span>, increasing a running-sum
statistic when a gene is found in <span class="math inline">\(S\)</span>
and decreasing it when a gene is not found in <span class="math inline">\(S\)</span>. The magnitude of the increment depends
on the correlation of the gene with the phenotype. The <span class="math inline">\(ES\)</span> is the maximum deviation from zero
encountered in the random walk.</p>
<p>The second step is the estimation of significance level of <span class="math inline">\(ES\)</span>. The statistical significance (nominal
p-value) of the <span class="math inline">\(ES\)</span> is estimated by
using an empirical phenotype-based permutation test that preserves the
complex correlation structure of the gene expression data. The phenotype
labels (<span class="math inline">\(L\)</span>) are permuted and the
<span class="math inline">\(ES\)</span> of the <span class="math inline">\(S\)</span> is recomputed for the permuted data,
which generates a null distribution for the <span class="math inline">\(ES\)</span>. The empirical, nominal p-value of the
observed <span class="math inline">\(ES\)</span> is then calculated
relative to this null distribution. The permutation of class labels
(groups) preserves gene-gene correlations and, thus, provides a more
biologically reasonable assessment of significance than would be
obtained by permuting genes.</p>
<p>Finally, the third step consist on the adjustment for multiple
hypothesis testing. When an entire database of gene sets is evaluated,
the estimated significance level is adjusted for multiple hypothesis
testing. First, the <span class="math inline">\(ES\)</span> is
normalized for each gene set to account for the size of the set,
yielding a normalized enrichment score (NES). Then, the proportion of
false positives is controlled by calculating the FDR corresponding to
each NES.</p>
<p>In 2010, a modification of the GSEA methodology was presented for
metabolomics studies. This method was called Metabolite Set Enrichment
Analysis (MSEA) and its main aim was to help researchers identify and
interpret patterns of human and mammalian metabolite concentration
changes in a biologically meaningful context <span class="citation">(<a href="#ref-xia2010msea">Xia and Wishart 2010</a>)</span>. MSEA is
currently widely used in the metabolomics community and it is
implemented and freely available at the known <a href="https://www.metaboanalyst.ca" class="external-link">MetaboAnalyst</a> web-based tool
<span class="citation">(<a href="#ref-xia2010msea">Xia and Wishart
2010</a>)</span>.</p>
<p>As can be seen, GSEA approach is more complex than the ORA
methodology, both in terms of methodological aspects and understanding
of the method.</p>
<p>The <code>fobitools</code> package provides a function to perform
MSEA using the FOBI information. This function requires a ranked list.
Here, we will use the metabolites obtained in the limma model ranked by
raw p-values.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">limma_FOBI_msea</span> <span class="op">&lt;-</span> <span class="va">limma_FOBI_names</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">FOBI</span>, <span class="va">pvalue</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">FOBI</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="op">-</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">pvalue</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">FOBI_msea</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="va">limma_FOBI_msea</span><span class="op">$</span><span class="va">pvalue</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">FOBI_msea</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">limma_FOBI_msea</span><span class="op">$</span><span class="va">FOBI</span></span>
<span></span>
<span><span class="va">msea_res</span> <span class="op">&lt;-</span> <span class="fu">fobitools</span><span class="fu">::</span><span class="fu"><a href="../reference/msea.html">msea</a></span><span class="op">(</span><span class="va">FOBI_msea</span>, pvalCutoff <span class="op">=</span> <span class="fl">0.06</span><span class="op">)</span></span>
<span></span>
<span><span class="va">msea_res</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kbl.html" class="external-link">kbl</a></span><span class="op">(</span>row.names <span class="op">=</span> <span class="cn">FALSE</span>, booktabs <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html" class="external-link">kable_styling</a></span><span class="op">(</span>latex_options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"striped"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
className
</th>
<th style="text-align:right;">
classSize
</th>
<th style="text-align:right;">
log2err
</th>
<th style="text-align:right;">
ES
</th>
<th style="text-align:right;">
NES
</th>
<th style="text-align:right;">
pval
</th>
<th style="text-align:right;">
padj
</th>
<th style="text-align:left;">
leadingEdge
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
berry (whole, raw)
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.2572065
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
2.021796
</td>
<td style="text-align:right;">
0.030969
</td>
<td style="text-align:right;">
0.8891762
</td>
<td style="text-align:left;">
FOBI:030590
</td>
</tr>
<tr>
<td style="text-align:left;">
butter
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.2572065
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
2.021796
</td>
<td style="text-align:right;">
0.030969
</td>
<td style="text-align:right;">
0.8891762
</td>
<td style="text-align:left;">
FOBI:030590
</td>
</tr>
<tr>
<td style="text-align:left;">
peanut butter
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.2572065
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
2.021796
</td>
<td style="text-align:right;">
0.030969
</td>
<td style="text-align:right;">
0.8891762
</td>
<td style="text-align:left;">
FOBI:030590
</td>
</tr>
</tbody>
</table>
<p>As we can see, the enrichment analysis with the MSEA method seems to
be much more accurate than the ORA method, since the two classes that
head the results table (“<em>grape (whole, raw)</em>” and
“<em>grapefruit (whole, raw)</em>”) are clearly within the FOBI food
group (set) “<em>plant fruit food product</em>”, which is aligned with
the study intervention, cranberry juice intake.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/fobi_graph.html">fobi_graph</a></span><span class="op">(</span>terms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FOODON:03301123"</span>, <span class="st">"FOODON:03301702"</span><span class="op">)</span>, </span>
<span>           get <span class="op">=</span> <span class="st">"anc"</span>, </span>
<span>           labels <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>           labelsize <span class="op">=</span> <span class="fl">6</span><span class="op">)</span></span></code></pre></div>
<p><img src="MW_ST000291_enrichment_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
<div class="section level4">
<h4 id="msea-plot-with-ggplot2">MSEA plot with <code>ggplot2</code><a class="anchor" aria-label="anchor" href="#msea-plot-with-ggplot2"></a>
</h4>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">msea_res</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">pval</span><span class="op">)</span>, y <span class="op">=</span> <span class="va">NES</span>, color <span class="op">=</span> <span class="va">NES</span>, size <span class="op">=</span> <span class="va">classSize</span>, label <span class="op">=</span> <span class="va">className</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"-log10(P-value)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"NES (Normalized Enrichment Score)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggrepel</span><span class="fu">::</span><span class="fu"><a href="https://ggrepel.slowkow.com/reference/geom_text_repel.html" class="external-link">geom_label_repel</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"black"</span>, size <span class="op">=</span> <span class="fl">7</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"top"</span>,</span>
<span>        text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">22</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html" class="external-link">scale_color_viridis_c</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_size.html" class="external-link">scale_size</a></span><span class="op">(</span>guide <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span></code></pre></div>
<p><img src="MW_ST000291_enrichment_files/figure-html/unnamed-chunk-16-1.png" width="1152"></p>
</div>
<div class="section level4">
<h4 id="network-of-metabolites-found-in-msea">Network of metabolites found in MSEA<a class="anchor" aria-label="anchor" href="#network-of-metabolites-found-in-msea"></a>
</h4>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">FOBI_terms</span> <span class="op">&lt;-</span> <span class="va">msea_res</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/unnest.html" class="external-link">unnest</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="va">leadingEdge</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">fobitools</span><span class="fu">::</span><span class="va"><a href="../reference/fobi.html">fobi</a></span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">FOBI</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">FOBI_terms</span><span class="op">$</span><span class="va">leadingEdge</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">id_code</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/fobi_graph.html">fobi_graph</a></span><span class="op">(</span>get <span class="op">=</span> <span class="st">"anc"</span>, </span>
<span>             labels <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>             legend <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>             labelsize <span class="op">=</span> <span class="fl">6</span>, </span>
<span>             legendSize <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<p><img src="MW_ST000291_enrichment_files/figure-html/unnamed-chunk-17-1.png" width="1152"></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="limitations">Limitations<a class="anchor" aria-label="anchor" href="#limitations"></a>
</h2>
<p>The FOBI ontology is currently in its first release version, so it
does not yet include information on many metabolites and food
relationships. All future efforts will be directed at expanding this
ontology, leading to a significant increase in the number of metabolites
and metabolite-food relationships. The <code>fobitools</code> package
provides the methodology for easy use of the FOBI ontology regardless of
the amount of information it contains. Therefore, future FOBI
improvements will also have a direct impact on the
<code>fobitools</code> package, increasing its utility and allowing to
perform, among others, more accurate, complete and robust enrichment
analyses.</p>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.2 (2023-10-31)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin20 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: macOS Sonoma 14.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: America/New_York</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] SummarizedExperiment_1.32.0   Biobase_2.62.0               </span></span>
<span><span class="co">#&gt;  [3] GenomicRanges_1.54.1          GenomeInfoDb_1.38.5          </span></span>
<span><span class="co">#&gt;  [5] IRanges_2.36.0                S4Vectors_0.40.2             </span></span>
<span><span class="co">#&gt;  [7] BiocGenerics_0.48.1           MatrixGenerics_1.14.0        </span></span>
<span><span class="co">#&gt;  [9] matrixStats_1.2.0             metabolomicsWorkbenchR_1.12.0</span></span>
<span><span class="co">#&gt; [11] POMA_1.13.13                  kableExtra_1.3.4             </span></span>
<span><span class="co">#&gt; [13] ggrepel_0.9.5                 rvest_1.0.3                  </span></span>
<span><span class="co">#&gt; [15] lubridate_1.9.3               forcats_1.0.0                </span></span>
<span><span class="co">#&gt; [17] stringr_1.5.1                 dplyr_1.1.4                  </span></span>
<span><span class="co">#&gt; [19] purrr_1.0.2                   readr_2.1.4                  </span></span>
<span><span class="co">#&gt; [21] tidyr_1.3.0                   tibble_3.2.1                 </span></span>
<span><span class="co">#&gt; [23] ggplot2_3.4.4                 tidyverse_2.0.0              </span></span>
<span><span class="co">#&gt; [25] fobitools_1.11.2              BiocStyle_2.30.0             </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] rstudioapi_0.15.0           jsonlite_1.8.8             </span></span>
<span><span class="co">#&gt;   [3] MultiAssayExperiment_1.28.0 magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;   [5] farver_2.1.1                rmarkdown_2.25             </span></span>
<span><span class="co">#&gt;   [7] zlibbioc_1.48.0             fs_1.6.3                   </span></span>
<span><span class="co">#&gt;   [9] ragg_1.2.7                  vctrs_0.6.5                </span></span>
<span><span class="co">#&gt;  [11] memoise_2.0.1               RCurl_1.98-1.13            </span></span>
<span><span class="co">#&gt;  [13] janitor_2.2.0               webshot_0.5.5              </span></span>
<span><span class="co">#&gt;  [15] S4Arrays_1.2.0              htmltools_0.5.7            </span></span>
<span><span class="co">#&gt;  [17] curl_5.2.0                  qdapRegex_0.7.8            </span></span>
<span><span class="co">#&gt;  [19] tictoc_1.2                  SparseArray_1.2.3          </span></span>
<span><span class="co">#&gt;  [21] sass_0.4.8                  parallelly_1.36.0          </span></span>
<span><span class="co">#&gt;  [23] bslib_0.6.1                 desc_1.4.3                 </span></span>
<span><span class="co">#&gt;  [25] impute_1.76.0               RecordLinkage_0.4-12.4     </span></span>
<span><span class="co">#&gt;  [27] cachem_1.0.8                igraph_1.6.0               </span></span>
<span><span class="co">#&gt;  [29] lifecycle_1.0.4             pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt;  [31] Matrix_1.6-1                R6_2.5.1                   </span></span>
<span><span class="co">#&gt;  [33] fastmap_1.1.1               snakecase_0.11.1           </span></span>
<span><span class="co">#&gt;  [35] GenomeInfoDbData_1.2.11     future_1.33.1              </span></span>
<span><span class="co">#&gt;  [37] selectr_0.4-2               digest_0.6.34              </span></span>
<span><span class="co">#&gt;  [39] colorspace_2.1-0            textshaping_0.3.7          </span></span>
<span><span class="co">#&gt;  [41] RSQLite_2.3.4               labeling_0.4.3             </span></span>
<span><span class="co">#&gt;  [43] fansi_1.0.6                 timechange_0.2.0           </span></span>
<span><span class="co">#&gt;  [45] abind_1.4-5                 httr_1.4.7                 </span></span>
<span><span class="co">#&gt;  [47] polyclip_1.10-6             compiler_4.3.2             </span></span>
<span><span class="co">#&gt;  [49] proxy_0.4-27                bit64_4.0.5                </span></span>
<span><span class="co">#&gt;  [51] withr_2.5.2                 BiocParallel_1.36.0        </span></span>
<span><span class="co">#&gt;  [53] viridis_0.6.4               DBI_1.2.0                  </span></span>
<span><span class="co">#&gt;  [55] highr_0.10                  ggforce_0.4.1              </span></span>
<span><span class="co">#&gt;  [57] MASS_7.3-60                 lava_1.7.3                 </span></span>
<span><span class="co">#&gt;  [59] DelayedArray_0.28.0         textclean_0.9.3            </span></span>
<span><span class="co">#&gt;  [61] tools_4.3.2                 future.apply_1.11.1        </span></span>
<span><span class="co">#&gt;  [63] nnet_7.3-19                 glue_1.7.0                 </span></span>
<span><span class="co">#&gt;  [65] grid_4.3.2                  fgsea_1.28.0               </span></span>
<span><span class="co">#&gt;  [67] generics_0.1.3              gtable_0.3.4               </span></span>
<span><span class="co">#&gt;  [69] tzdb_0.4.0                  class_7.3-22               </span></span>
<span><span class="co">#&gt;  [71] data.table_1.14.10          hms_1.1.3                  </span></span>
<span><span class="co">#&gt;  [73] tidygraph_1.3.0             xml2_1.3.6                 </span></span>
<span><span class="co">#&gt;  [75] utf8_1.2.4                  XVector_0.42.0             </span></span>
<span><span class="co">#&gt;  [77] pillar_1.9.0                limma_3.58.1               </span></span>
<span><span class="co">#&gt;  [79] vroom_1.6.5                 splines_4.3.2              </span></span>
<span><span class="co">#&gt;  [81] tweenr_2.0.2                lattice_0.22-5             </span></span>
<span><span class="co">#&gt;  [83] survival_3.5-7              bit_4.0.5                  </span></span>
<span><span class="co">#&gt;  [85] tidyselect_1.2.0            knitr_1.45                 </span></span>
<span><span class="co">#&gt;  [87] gridExtra_2.3               bookdown_0.37              </span></span>
<span><span class="co">#&gt;  [89] svglite_2.1.3               xfun_0.41                  </span></span>
<span><span class="co">#&gt;  [91] graphlayouts_1.0.2          statmod_1.5.0              </span></span>
<span><span class="co">#&gt;  [93] stringi_1.8.3               yaml_2.3.8                 </span></span>
<span><span class="co">#&gt;  [95] evaluate_0.23               codetools_0.2-19           </span></span>
<span><span class="co">#&gt;  [97] evd_2.3-6.1                 ggraph_2.1.0               </span></span>
<span><span class="co">#&gt;  [99] BiocManager_1.30.22         cli_3.6.2                  </span></span>
<span><span class="co">#&gt; [101] ontologyIndex_2.11          rpart_4.1.23               </span></span>
<span><span class="co">#&gt; [103] xtable_1.8-4                systemfonts_1.0.5          </span></span>
<span><span class="co">#&gt; [105] struct_1.14.0               munsell_0.5.0              </span></span>
<span><span class="co">#&gt; [107] jquerylib_0.1.4             Rcpp_1.0.12                </span></span>
<span><span class="co">#&gt; [109] globals_0.16.2              parallel_4.3.2             </span></span>
<span><span class="co">#&gt; [111] pkgdown_2.0.7               blob_1.2.4                 </span></span>
<span><span class="co">#&gt; [113] bitops_1.0-7                ff_4.0.12                  </span></span>
<span><span class="co">#&gt; [115] listenv_0.9.0               viridisLite_0.4.2          </span></span>
<span><span class="co">#&gt; [117] ipred_0.9-14                scales_1.3.0               </span></span>
<span><span class="co">#&gt; [119] prodlim_2023.08.28          e1071_1.7-14               </span></span>
<span><span class="co">#&gt; [121] crayon_1.5.2                clisymbols_1.2.0           </span></span>
<span><span class="co">#&gt; [123] rlang_1.1.3                 ada_2.0-5                  </span></span>
<span><span class="co">#&gt; [125] cowplot_1.1.2               fastmatch_1.1-4</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-POMA" class="csl-entry">
Castellano-Escuder, Pol, Cristina Andrés-Lacueva, and Alex Sánchez-Pla.
2021. <em>POMA: User-Friendly Workflow for Pre-Processing and
Statistical Analysis of Mass Spectrometry Data</em>. <a href="https://github.com/pcastellanoescuder/POMA" class="external-link">https://github.com/pcastellanoescuder/POMA</a>.
</div>
<div id="ref-castellano2020fobi" class="csl-entry">
Castellano-Escuder, Pol, Raúl González-Domı́nguez, David S Wishart,
Cristina Andrés-Lacueva, and Alex Sánchez-Pla. 2020. <span>“FOBI: An
Ontology to Represent Food Intake Data and Associate It with Metabolomic
Data.”</span> <em>Database</em> 2020.
</div>
<div id="ref-creixell2015pathway" class="csl-entry">
Creixell, Pau, Jüri Reimand, Syed Haider, Guanming Wu, Tatsuhiro
Shibata, Miguel Vazquez, Ville Mustonen, et al. 2015. <span>“Pathway and
Network Analysis of Cancer Genomes.”</span> <em>Nature Methods</em> 12
(7): 615.
</div>
<div id="ref-limma" class="csl-entry">
Ritchie, Matthew E, Belinda Phipson, Di Wu, Yifang Hu, Charity W Law,
Wei Shi, and Gordon K Smyth. 2015. <span>“<span class="nocase">limma</span> Powers Differential Expression Analyses for
<span>RNA</span>-Sequencing and Microarray Studies.”</span> <em>Nucleic
Acids Research</em> 43 (7): e47. <a href="https://doi.org/10.1093/nar/gkv007" class="external-link">https://doi.org/10.1093/nar/gkv007</a>.
</div>
<div id="ref-subramanian2005gene" class="csl-entry">
Subramanian, Aravind, Pablo Tamayo, Vamsi K Mootha, Sayan Mukherjee,
Benjamin L Ebert, Michael A Gillette, Amanda Paulovich, et al. 2005.
<span>“Gene Set Enrichment Analysis: A Knowledge-Based Approach for
Interpreting Genome-Wide Expression Profiles.”</span> <em>Proceedings of
the National Academy of Sciences</em> 102 (43): 15545–50.
</div>
<div id="ref-sud2016metabolomics" class="csl-entry">
Sud, Manish, Eoin Fahy, Dawn Cotter, Kenan Azam, Ilango Vadivelu,
Charles Burant, Arthur Edison, et al. 2016. <span>“Metabolomics
Workbench: An International Repository for Metabolomics Data and
Metadata, Metabolite Standards, Protocols, Tutorials and Training, and
Analysis Tools.”</span> <em>Nucleic Acids Research</em> 44 (D1):
D463–70.
</div>
<div id="ref-xia2010msea" class="csl-entry">
Xia, Jianguo, and David S Wishart. 2010. <span>“MSEA: A Web-Based Tool
to Identify Biologically Meaningful Patterns in Quantitative Metabolomic
Data.”</span> <em>Nucleic Acids Research</em> 38 (suppl_2): W71–77.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Pol Castellano-Escuder, Alex Sánchez-Pla.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
