<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Use case of fobitools using tidyverse and the ST000291 data from Metabolomics Workbench • fobitools</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/spacelab/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Use case of fobitools using tidyverse and the ST000291 data from Metabolomics Workbench">
<meta property="og:description" content="fobitools">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">fobitools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.56</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/MW_ST000291_enrichment.html">Use case ST000291
</a>
    </li>
    <li>
      <a href="../articles/MW_ST000629_enrichment.html">Use case ST000629
</a>
    </li>
    <li>
      <a href="../articles/food_enrichment_analysis.html">Simple food ORA
</a>
    </li>
    <li>
      <a href="../articles/dietary_data_annotation.html">Dietary text annotation
</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/pcastellanoescuder/fobitools">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="MW_ST000291_enrichment_files/accessible-code-block-0.0.1/empty-anchor.js"></script><script src="MW_ST000291_enrichment_files/kePrint-0.0.1/kePrint.js"></script><link href="MW_ST000291_enrichment_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Use case of fobitools using tidyverse and the ST000291 data from Metabolomics Workbench</h1>
                        <h4 class="author">Pol Castellano-Escuder</h4>
            <address class="author_afil">
      University of Barcelona, Spain.<br><a class="author_email" href="mailto:#"></a><a href="mailto:polcaes@gmail.com" class="email">polcaes@gmail.com</a>
      </address>
                  
            <h4 class="date">14 mayo 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pcastellanoescuder/fobitools/blob/master/vignettes/MW_ST000291_enrichment.Rmd"><code>vignettes/MW_ST000291_enrichment.Rmd</code></a></small>
      <div class="hidden name"><code>MW_ST000291_enrichment.Rmd</code></div>

    </div>

    
    
<p><strong>Compiled date</strong>: 2021-05-14</p>
<p><strong>Last edited</strong>: 2021-10-05</p>
<p><strong>License</strong>: GPL-3</p>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>Run the following code to install the Bioconductor version of the package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>

<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"fobitools"</span><span class="op">)</span></code></pre></div>
</div>
<div id="load-packages" class="section level1">
<h1 class="hasAnchor">
<a href="#load-packages" class="anchor"></a>Load packages</h1>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/pcastellanoescuder/fobitools/">fobitools</a></span><span class="op">)</span></code></pre></div>
<p>We will also need some additional <a href="https://cran.r-project.org">CRAN</a> and <a href="https://bioconductor.org">Bioconductor</a> packages for performing tasks such as statistical analysis and web scraping.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># CRAN</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://rvest.tidyverse.org/">rvest</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/slowkow/ggrepel">ggrepel</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://haozhu233.github.io/kableExtra/">kableExtra</a></span><span class="op">)</span>

<span class="co"># Bioconductor</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/pcastellanoescuder/POMA">POMA</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">metabolomicsWorkbenchR</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment">SummarizedExperiment</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="download-the-data-from-metabolomics-workbench" class="section level1">
<h1 class="hasAnchor">
<a href="#download-the-data-from-metabolomics-workbench" class="anchor"></a>Download the data from Metabolomics Workbench</h1>
<p>The Metabolomics Workbench, available at <a href="www.metabolomicsworkbench.org">www.metabolomicsworkbench.org</a>, is a public repository for metabolomics metadata and experimental data spanning various species and experimental platforms, metabolite standards, metabolite structures, protocols, tutorials, and training material and other educational resources. It provides a computational platform to integrate, analyze, track, deposit and disseminate large volumes of heterogeneous data from a wide variety of metabolomics studies including mass spectrometry (MS) and nuclear magnetic resonance spectrometry (NMR) data spanning over 20 different species covering all the major taxonomic categories including humans and other mammals, plants, insects, invertebrates and microorganisms <span class="citation">(Sud et al. <a href="#ref-sud2016metabolomics" role="doc-biblioref">2016</a>)</span>.</p>
<p>The <code>metabolomicsWorkbenchR</code> Bioconductor package allows us to obtain data from the Metabolomics Workbench repository. In this vignette we will use the sample data set <a href="https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Study&amp;StudyID=ST000291">ST000291</a>.</p>
<div id="sumary-of-the-study-st000291" class="section level2">
<h2 class="hasAnchor">
<a href="#sumary-of-the-study-st000291" class="anchor"></a>Sumary of the study (ST000291)</h2>
<p>Eighteen healthy female college students between 21-29 years old with a normal BMI of 18.5-25 were recruited. Each subject was provided with a list of foods that contained significant amount of procyanidins, such as <strong>cranberries, apples, grapes, blueberries, chocolate and plums</strong>. They were advised to avoid these foods during the 1-6th day and the rest of the study. On the morning of the 7th day, a first-morning baseline urine sample and blood sample were collected from all human subjects after overnight fasting. <strong>Participants were then randomly allocated into two groups (n=9) to consume cranberry juice or apple juice</strong>. Six bottles (250 ml/bottle) of juice were given to participants to drink in the morning and evening of the 7th, 8th, and 9th day. On the morning of 10th day, all subjects returned to the clinical unit to provide a first-morning urine sample after overnight fasting. The blood sample was also collected from participants 30 min later after they drank another bottle of juice in the morning. After two-weeks of wash out period, participants switched to the alternative regimen and repeated the protocol. One human subject was dropped off this study because she missed part of her appointments. Another two human subjects were removed from urine metabolomics analyses because they failed to provide required urine samples after juice drinking.The present study aimed to investigate overall metabolic changes caused by procyanidins concentrates from cranberries and apples using a global LCMS based metabolomics approach. All plasma and urine samples were stored at -80ºC until analysis.</p>
</div>
<div id="download-data" class="section level2">
<h2 class="hasAnchor">
<a href="#download-data" class="anchor"></a>Download data</h2>
<p>This study is composed of two complementary MS analyses, the positive mode (AN000464) and the negative mode (AN000465). Let’s download them both!</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data_negative_mode</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/metabolomicsWorkbenchR/man/do_query.html">do_query</a></span><span class="op">(</span>
  context <span class="op">=</span> <span class="st">"study"</span>,
  input_item <span class="op">=</span> <span class="st">"analysis_id"</span>,
  input_value <span class="op">=</span> <span class="st">"AN000465"</span>,
  output_item <span class="op">=</span> <span class="st">"SummarizedExperiment"</span><span class="op">)</span>

<span class="va">data_positive_mode</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/metabolomicsWorkbenchR/man/do_query.html">do_query</a></span><span class="op">(</span>
  context <span class="op">=</span> <span class="st">"study"</span>,
  input_item <span class="op">=</span> <span class="st">"analysis_id"</span>,
  input_value <span class="op">=</span> <span class="st">"AN000464"</span>,
  output_item <span class="op">=</span> <span class="st">"SummarizedExperiment"</span><span class="op">)</span></code></pre></div>
</div>
<div id="scraping-metabolite-names-and-identifiers-with-rvest" class="section level2">
<h2 class="hasAnchor">
<a href="#scraping-metabolite-names-and-identifiers-with-rvest" class="anchor"></a>Scraping metabolite names and identifiers with <code>rvest</code>
</h2>
<p>In many metabolomics studies, the reproducibility of analyses is severely affected by the poor interoperability of metabolite names and their identifiers. For this reason it is important to develop tools that facilitate the process of converting one type of identifier to another. In order to use the <code>fobitools</code> package, we need some generic identifier (such as PubChem, KEGG or HMDB) that allows us to obtain the corresponding FOBI identifier for each metabolite. The Metabolomics Workbench repository provides us with this information for many of the metabolites quantified in study <a href="https://www.metabolomicsworkbench.org/data/show_metabolites_by_study.php?STUDY_ID=ST000291&amp;SEARCH_TYPE=KNOWN&amp;STUDY_TYPE=MS&amp;RESULT_TYPE=1">ST000291</a> (Figure @ref(fig:metabolitenames)). In order to easily obtain this information, we will perform a web scraping operation using the <code>rvest</code> package.</p>
<div class="figure" style="text-align: center">
<img src="figure/ST000291_names.png" alt="Metabolite identifiers of the ST000291 Metabolomics Workbench study." width="100%"><p class="caption">
Metabolite identifiers of the ST000291 Metabolomics Workbench study.
</p>
</div>
<p>Below we obtain the PubChem and KEGG identifiers of the metabolites analyzed in the positive and negative mode directly from the Metabolomics Workbench website. We will then remove those duplicate identifiers.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">metaboliteNamesURL</span> <span class="op">&lt;-</span> <span class="st">"https://www.metabolomicsworkbench.org/data/show_metabolites_by_study.php?STUDY_ID=ST000291&amp;SEARCH_TYPE=KNOWN&amp;STUDY_TYPE=MS&amp;RESULT_TYPE=1"</span>
<span class="va">metaboliteNames</span> <span class="op">&lt;-</span> <span class="va">metaboliteNamesURL</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="http://xml2.r-lib.org/reference/read_xml.html">read_html</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://rvest.tidyverse.org/reference/rename.html">html_nodes</a></span><span class="op">(</span><span class="st">".datatable"</span><span class="op">)</span>

<span class="va">metaboliteNames_negative</span> <span class="op">&lt;-</span> <span class="va">metaboliteNames</span> <span class="op">%&gt;%</span>
  <span class="va">.</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rvest.tidyverse.org/reference/html_table.html">html_table</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">`Metabolite Name`</span>, <span class="va">PubChemCompound_ID</span>, <span class="va">`Kegg Id`</span><span class="op">)</span>

<span class="va">metaboliteNames_positive</span> <span class="op">&lt;-</span> <span class="va">metaboliteNames</span> <span class="op">%&gt;%</span>
  <span class="va">.</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rvest.tidyverse.org/reference/html_table.html">html_table</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">`Metabolite Name`</span>, <span class="va">PubChemCompound_ID</span>, <span class="va">`Kegg Id`</span><span class="op">)</span>

<span class="va">metaboliteNames</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">metaboliteNames_negative</span>, <span class="va">metaboliteNames_positive</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span>names <span class="op">=</span> <span class="fl">1</span>, PubChem <span class="op">=</span> <span class="fl">2</span>, KEGG <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">mutate</span><span class="op">(</span>KEGG <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">KEGG</span> <span class="op">==</span> <span class="st">"-"</span>, <span class="st">"UNKNOWN"</span>, <span class="va">KEGG</span><span class="op">)</span>,
         PubChem <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span><span class="op">(</span><span class="va">PubChem</span> <span class="op">==</span> <span class="st">"-"</span>, <span class="st">"UNKNOWN"</span>, <span class="va">PubChem</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/duplicated.html">duplicated</a></span><span class="op">(</span><span class="va">PubChem</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="prepare-features-and-metadata" class="section level1">
<h1 class="hasAnchor">
<a href="#prepare-features-and-metadata" class="anchor"></a>Prepare features and metadata</h1>
<p>Now we have to prepare the metadata and features in order to proceed with the statistical analysis. In this step we assign to each metabolite its PubChem identifier obtained in the previous step.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## negative mode features </span>
<span class="va">features_negative</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">data_negative_mode</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html">slice</a></span><span class="op">(</span><span class="op">-</span><span class="fu">n</span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">features_negative</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">data_negative_mode</span><span class="op">)</span><span class="op">$</span><span class="va">metabolite</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">data_negative_mode</span><span class="op">)</span><span class="op">$</span><span class="va">metabolite</span><span class="op">)</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span><span class="op">]</span>

<span class="co">## positive mode features</span>
<span class="va">features_positive</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">data_positive_mode</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html">slice</a></span><span class="op">(</span><span class="op">-</span><span class="fu">n</span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">features_positive</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">data_positive_mode</span><span class="op">)</span><span class="op">$</span><span class="va">metabolite</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">data_positive_mode</span><span class="op">)</span><span class="op">$</span><span class="va">metabolite</span><span class="op">)</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span><span class="op">]</span>

<span class="co">## combine positive and negative mode and set PubChem IDs as feature names</span>
<span class="va">features</span> <span class="op">&lt;-</span> <span class="fu">bind_rows</span><span class="op">(</span><span class="va">features_negative</span>, <span class="va">features_positive</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"names"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">right_join</span><span class="op">(</span><span class="va">metaboliteNames</span>, by <span class="op">=</span> <span class="st">"names"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">names</span>, <span class="op">-</span><span class="va">KEGG</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"PubChem"</span><span class="op">)</span>

<span class="co">## metadata</span>
<span class="va">pdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">data_negative_mode</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="co"># or "data_positive_mode". They are equal</span>
  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"ID"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">mutate</span><span class="op">(</span>Treatment <span class="op">=</span> <span class="fu">case_when</span><span class="op">(</span><span class="va">Treatment</span> <span class="op">==</span> <span class="st">"Baseline urine"</span> <span class="op">~</span> <span class="st">"Baseline"</span>,
                               <span class="va">Treatment</span> <span class="op">==</span> <span class="st">"Urine after drinking apple juice"</span> <span class="op">~</span> <span class="st">"Apple"</span>,
                               <span class="va">Treatment</span> <span class="op">==</span> <span class="st">"Urine after drinking cranberry juice"</span> <span class="op">~</span> <span class="st">"Cranberry"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="statistical-analysis-with-poma" class="section level1">
<h1 class="hasAnchor">
<a href="#statistical-analysis-with-poma" class="anchor"></a>Statistical analysis with <code>POMA</code>
</h1>
<p><code>POMA</code> provides a structured, reproducible and easy-to-use workflow for the visualization, preprocessing, exploration, and statistical analysis of metabolomics and proteomics data. The main aim of this package is to enable a flexible data cleaning and statistical analysis processes in one comprehensible and user-friendly R package. <code>POMA</code> uses the standardized <code>MSnbase</code> data structures, to achieve the maximum flexibility and reproducibility and makes <code>POMA</code> compatible with other Bioconductor packages <span class="citation">(Castellano-Escuder, Andrés-Lacueva, and Sánchez-Pla <a href="#ref-POMA" role="doc-biblioref">2021</a>)</span>.</p>
<div id="create-a-msnbasemsnset-object" class="section level2">
<h2 class="hasAnchor">
<a href="#create-a-msnbasemsnset-object" class="anchor"></a>Create a <code>MSnbase::MSnSet</code> object</h2>
<p>First, we create a <code>MSnSet</code> object that integrates both metadata and features in the same data structure.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data_msnset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/POMA/man/PomaMSnSetClass.html">PomaMSnSetClass</a></span><span class="op">(</span>target <span class="op">=</span> <span class="va">pdata</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">features</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="preprocessing" class="section level2">
<h2 class="hasAnchor">
<a href="#preprocessing" class="anchor"></a>Preprocessing</h2>
<p>Second, we perform the preprocessing step. This step includes the missing value imputation unsing the <span class="math inline">\(k\)</span>-NN algorithm, log Pareto normalization (transformation and scaling) and outlier detection and cleaning. Once these steps are completed, we can proceed to the statistical analysis of these data.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data_preprocessed</span> <span class="op">&lt;-</span> <span class="va">data_msnset</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/POMA/man/PomaImpute.html">PomaImpute</a></span><span class="op">(</span>ZerosAsNA <span class="op">=</span> <span class="cn">TRUE</span>, cutoff <span class="op">=</span> <span class="fl">20</span>, method <span class="op">=</span> <span class="st">"knn"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/POMA/man/PomaNorm.html">PomaNorm</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"log_pareto"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/POMA/man/PomaOutliers.html">PomaOutliers</a></span><span class="op">(</span>coef <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></code></pre></div>
</div>
<div id="limma-model" class="section level2">
<h2 class="hasAnchor">
<a href="#limma-model" class="anchor"></a>Limma model</h2>
<p>We use a limma model <span class="citation">(Ritchie et al. <a href="#ref-limma" role="doc-biblioref">2015</a>)</span> to identify those most significant metabolites between the “<strong>Baseline urine</strong>” and “<strong>Urine after drinking cranberry juice</strong>” groups. With this analysis we expect to find metabolites related to cranberry intake.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">limma_res</span> <span class="op">&lt;-</span> <span class="va">data_preprocessed</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/POMA/man/PomaLimma.html">PomaLimma</a></span><span class="op">(</span>contrast <span class="op">=</span> <span class="st">"Baseline-Cranberry"</span>, adjust <span class="op">=</span> <span class="st">"fdr"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"PubChemCID"</span><span class="op">)</span>

<span class="co"># show the first 10 features</span>
<span class="va">limma_res</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html">slice</a></span><span class="op">(</span><span class="fl">1L</span><span class="op">:</span><span class="fl">10L</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kbl.html">kbl</a></span><span class="op">(</span>row.names <span class="op">=</span> <span class="cn">FALSE</span>, booktabs <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html">kable_styling</a></span><span class="op">(</span>latex_options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"striped"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table class="table table">
<thead><tr>
<th style="text-align:left;">
PubChemCID
</th>
<th style="text-align:right;">
logFC
</th>
<th style="text-align:right;">
AveExpr
</th>
<th style="text-align:right;">
t
</th>
<th style="text-align:right;">
P.Value
</th>
<th style="text-align:right;">
adj.P.Val
</th>
<th style="text-align:right;">
B
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
54678503
</td>
<td style="text-align:right;">
-2.015867
</td>
<td style="text-align:right;">
-2.22e-05
</td>
<td style="text-align:right;">
-7.319346
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:right;">
0.0000035
</td>
<td style="text-align:right;">
11.031428
</td>
</tr>
<tr>
<td style="text-align:left;">
94214
</td>
<td style="text-align:right;">
-1.436933
</td>
<td style="text-align:right;">
-4.44e-05
</td>
<td style="text-align:right;">
-5.720525
</td>
<td style="text-align:right;">
7.0e-07
</td>
<td style="text-align:right;">
0.0004658
</td>
<td style="text-align:right;">
5.827603
</td>
</tr>
<tr>
<td style="text-align:left;">
5378303
</td>
<td style="text-align:right;">
-1.591333
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
-5.578285
</td>
<td style="text-align:right;">
1.2e-06
</td>
<td style="text-align:right;">
0.0005079
</td>
<td style="text-align:right;">
5.367712
</td>
</tr>
<tr>
<td style="text-align:left;">
71485
</td>
<td style="text-align:right;">
-1.520600
</td>
<td style="text-align:right;">
4.44e-05
</td>
<td style="text-align:right;">
-5.417065
</td>
<td style="text-align:right;">
2.0e-06
</td>
<td style="text-align:right;">
0.0005552
</td>
<td style="text-align:right;">
4.848693
</td>
</tr>
<tr>
<td style="text-align:left;">
5486800
</td>
<td style="text-align:right;">
1.175733
</td>
<td style="text-align:right;">
0.00e+00
</td>
<td style="text-align:right;">
5.403865
</td>
<td style="text-align:right;">
2.1e-06
</td>
<td style="text-align:right;">
0.0005552
</td>
<td style="text-align:right;">
4.806317
</td>
</tr>
<tr>
<td style="text-align:left;">
17531
</td>
<td style="text-align:right;">
-1.490867
</td>
<td style="text-align:right;">
2.22e-05
</td>
<td style="text-align:right;">
-5.350936
</td>
<td style="text-align:right;">
2.6e-06
</td>
<td style="text-align:right;">
0.0005552
</td>
<td style="text-align:right;">
4.636603
</td>
</tr>
<tr>
<td style="text-align:left;">
3035199
</td>
<td style="text-align:right;">
-1.298667
</td>
<td style="text-align:right;">
-2.22e-05
</td>
<td style="text-align:right;">
-5.074320
</td>
<td style="text-align:right;">
6.6e-06
</td>
<td style="text-align:right;">
0.0011642
</td>
<td style="text-align:right;">
3.755554
</td>
</tr>
<tr>
<td style="text-align:left;">
439361
</td>
<td style="text-align:right;">
-1.061267
</td>
<td style="text-align:right;">
-2.22e-05
</td>
<td style="text-align:right;">
-5.030209
</td>
<td style="text-align:right;">
7.6e-06
</td>
<td style="text-align:right;">
0.0011642
</td>
<td style="text-align:right;">
3.616094
</td>
</tr>
<tr>
<td style="text-align:left;">
5353
</td>
<td style="text-align:right;">
-1.349133
</td>
<td style="text-align:right;">
4.44e-05
</td>
<td style="text-align:right;">
-5.010060
</td>
<td style="text-align:right;">
8.2e-06
</td>
<td style="text-align:right;">
0.0011642
</td>
<td style="text-align:right;">
3.552497
</td>
</tr>
<tr>
<td style="text-align:left;">
1132
</td>
<td style="text-align:right;">
-1.579533
</td>
<td style="text-align:right;">
2.22e-05
</td>
<td style="text-align:right;">
-4.983350
</td>
<td style="text-align:right;">
8.9e-06
</td>
<td style="text-align:right;">
0.0011642
</td>
<td style="text-align:right;">
3.468300
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="convert-pubchem-ids-to-fobi-ids" class="section level1">
<h1 class="hasAnchor">
<a href="#convert-pubchem-ids-to-fobi-ids" class="anchor"></a>Convert PubChem IDs to FOBI IDs</h1>
<p>Once we have the results of the statistical analysis and generic identifiers recognized in the FOBI ontology <span class="citation">(Castellano-Escuder et al. <a href="#ref-castellano2020fobi" role="doc-biblioref">2020</a>)</span>, we can proceed to perform one of the main functions provided by the <code>fobitools</code> package, the ID conversion. With the <code><a href="../reference/id_convert.html">fobitools::id_convert()</a></code> command, users can convert different IDs between FOBI, HMDB, KEGG, PubChem, InChIKey, InChICode, ChemSpider, and chemical names. We will then obtain the FOBI IDs from the PubChem IDs (obtained in the previous sections) and add them as a new column to the results of the limma model.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">limma_FOBI_names</span> <span class="op">&lt;-</span> <span class="va">limma_res</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="st">"PubChemCID"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">fobitools</span><span class="fu">::</span><span class="fu"><a href="../reference/id_convert.html">id_convert</a></span><span class="op">(</span><span class="op">)</span>

<span class="co"># show the ID conversion results</span>
<span class="va">limma_FOBI_names</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kbl.html">kbl</a></span><span class="op">(</span>row.names <span class="op">=</span> <span class="cn">FALSE</span>, booktabs <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html">kable_styling</a></span><span class="op">(</span>latex_options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"striped"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table class="table table">
<thead><tr>
<th style="text-align:left;">
FOBI
</th>
<th style="text-align:left;">
PubChemCID
</th>
<th style="text-align:left;">
ChemSpider
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
FOBI:030415
</td>
<td style="text-align:left;">
91
</td>
<td style="text-align:left;">
89
</td>
</tr>
<tr>
<td style="text-align:left;">
FOBI:030711
</td>
<td style="text-align:left;">
1145
</td>
<td style="text-align:left;">
1113
</td>
</tr>
<tr>
<td style="text-align:left;">
FOBI:030555
</td>
<td style="text-align:left;">
5280445
</td>
<td style="text-align:left;">
4444102
</td>
</tr>
<tr>
<td style="text-align:left;">
FOBI:030709
</td>
<td style="text-align:left;">
1123
</td>
<td style="text-align:left;">
10675782
</td>
</tr>
<tr>
<td style="text-align:left;">
FOBI:030625
</td>
<td style="text-align:left;">
7533
</td>
<td style="text-align:left;">
15484224
</td>
</tr>
<tr>
<td style="text-align:left;">
FOBI:030397
</td>
<td style="text-align:left;">
1794427
</td>
<td style="text-align:left;">
1405788
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">limma_FOBI_names</span> <span class="op">&lt;-</span> <span class="va">limma_FOBI_names</span> <span class="op">%&gt;%</span>
  <span class="fu">right_join</span><span class="op">(</span><span class="va">limma_res</span>, by <span class="op">=</span> <span class="st">"PubChemCID"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="op">-</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html">desc</a></span><span class="op">(</span><span class="va">P.Value</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="enrichment-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#enrichment-analysis" class="anchor"></a>Enrichment analysis</h1>
<p>Enrichment analysis denotes any method that benefits from biological pathway or network information to gain insight into a biological system <span class="citation">(Creixell et al. <a href="#ref-creixell2015pathway" role="doc-biblioref">2015</a>)</span>. In other words, these type of analyses integrate the existing biological knowledge (from different biological sources such as databases and ontologies) and the statistical results of <em>omics</em> studies, obtaining a deeper understanding of biological systems.</p>
<p>In most metabolomics studies, the output of statistical analysis is usually a list of features selected as statistically significant or statistically relevant according to a pre-defined statistical criteria. Enrichment analysis methods use these selected features to explore associated biologically relevant pathways, diseases, etc., depending on the nature of the input feature list (genes, metabolites, etc.) and the source used to extract the biological knowledge (GO, KEGG, <strong>FOBI</strong>, etc.).</p>
<p>Here, we present a tool that uses the FOBI information to perform different types of enrichment analyses. Therefore, the presented methods allow researchers to move from lists of metabolites to chemical classes and food groups associated with those lists, and consequently, to the study design.</p>
<p>Currently, the most popular used approaches for enrichment analysis are the over representation analysis (ORA) and the gene set enrichment analysis (GSEA), with its variants for other fields such as the metabolite set enrichment analysis (MSEA) <span class="citation">(Xia and Wishart <a href="#ref-xia2010msea" role="doc-biblioref">2010</a>)</span>.</p>
<div id="over-representation-analysis-ora" class="section level2">
<h2 class="hasAnchor">
<a href="#over-representation-analysis-ora" class="anchor"></a>Over representation analysis (ORA)</h2>
<p>ORA is one of the most used methods to perform enrichment analysis in metabolomics studies due to its simplicity and easy understanding. This method statistically evaluates the fraction of metabolites in a particular pathway found among the set of metabolites statistically selected. Thus, ORA is used to test if certain groups of metabolites are represented more than expected by chance given a feature list.</p>
<p>However, ORA has a number of limitations. The most important one is the need of using a certain threshold or criteria to select the feature list. This means that metabolites do not meet the selection criteria must be discarded. The second big limitation of ORA is that this method assumes independence of sets and features. In ORA, is assumed that each feature is independent of the other features and each set is independent of the other sets.</p>
<p>Here, we perform an ORA with the <code>fobitools</code> package, where we will use as a universe all the metabolites of the study present in FOBI and as a list those metabolites with a raw p-value &lt; 0.01 in the limma results table.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">metaboliteList</span> <span class="op">&lt;-</span> <span class="va">limma_FOBI_names</span><span class="op">$</span><span class="va">FOBI</span><span class="op">[</span><span class="va">limma_FOBI_names</span><span class="op">$</span><span class="va">P.Value</span> <span class="op">&lt;</span> <span class="fl">0.01</span><span class="op">]</span>
<span class="va">metaboliteUniverse</span> <span class="op">&lt;-</span> <span class="va">limma_FOBI_names</span><span class="op">$</span><span class="va">FOBI</span>

<span class="fu">fobitools</span><span class="fu">::</span><span class="fu"><a href="../reference/ora.html">ora</a></span><span class="op">(</span>metaboliteList <span class="op">=</span> <span class="va">metaboliteList</span>,
               metaboliteUniverse <span class="op">=</span> <span class="va">metaboliteUniverse</span>,
               pvalCutoff <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kbl.html">kbl</a></span><span class="op">(</span>row.names <span class="op">=</span> <span class="cn">FALSE</span>, booktabs <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html">kable_styling</a></span><span class="op">(</span>latex_options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"striped"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table class="table table">
<thead><tr>
<th style="text-align:left;">
className
</th>
<th style="text-align:right;">
classSize
</th>
<th style="text-align:right;">
overlap
</th>
<th style="text-align:right;">
pval
</th>
<th style="text-align:right;">
padj
</th>
<th style="text-align:left;">
overlapMetabolites
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
soft drink (dietetic)
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.1038407
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
FOBI:030627
</td>
</tr>
<tr>
<td style="text-align:left;">
olive oil
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.4623044
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:left;">
FOBI:030340
</td>
</tr>
</tbody>
</table>
<p>As we can see, due to the limitations of this methodology and the small number of metabolites that meet the set statistical criterion, the results do not show a clear and obvious relationship with the design of the study, as the food groups that appear in the ORA results do not correspond to those foods administered in the intervention.</p>
</div>
<div id="msea" class="section level2">
<h2 class="hasAnchor">
<a href="#msea" class="anchor"></a>MSEA</h2>
<p>Gene Set Enrichment Analysis (GSEA) methodology was proposed for the first time in 2005, with the aim of improving the interpretation of gene expression data. The main purpose of GSEA is to determine whether members of a gene set <span class="math inline">\(S\)</span> tend to occur toward the top (or bottom) of the gene list <span class="math inline">\(L\)</span>, in which case the gene set is correlated with the phenotypic class distinction <span class="citation">(Subramanian et al. <a href="#ref-subramanian2005gene" role="doc-biblioref">2005</a>)</span>.</p>
<p>This type of analysis basically consists of three key steps <span class="citation">(Subramanian et al. <a href="#ref-subramanian2005gene" role="doc-biblioref">2005</a>)</span>:</p>
<p>The first step consists on the calculation of an enrichment score (<span class="math inline">\(ES\)</span>). This value indicates the degree to which a set <span class="math inline">\(S\)</span> is overrepresented at the extremes (top or bottom) of the entire ranked gene list <span class="math inline">\(L\)</span>. The <span class="math inline">\(ES\)</span> is calculated by walking down the list <span class="math inline">\(L\)</span>, increasing a running-sum statistic when a gene is found in <span class="math inline">\(S\)</span> and decreasing it when a gene is not found in <span class="math inline">\(S\)</span>. The magnitude of the increment depends on the correlation of the gene with the phenotype. The <span class="math inline">\(ES\)</span> is the maximum deviation from zero encountered in the random walk.</p>
<p>The second step is the estimation of significance level of <span class="math inline">\(ES\)</span>. The statistical significance (nominal p-value) of the <span class="math inline">\(ES\)</span> is estimated by using an empirical phenotype-based permutation test that preserves the complex correlation structure of the gene expression data. The phenotype labels (<span class="math inline">\(L\)</span>) are permuted and the <span class="math inline">\(ES\)</span> of the <span class="math inline">\(S\)</span> is recomputed for the permuted data, which generates a null distribution for the <span class="math inline">\(ES\)</span>. The empirical, nominal p-value of the observed <span class="math inline">\(ES\)</span> is then calculated relative to this null distribution. The permutation of class labels (groups) preserves gene-gene correlations and, thus, provides a more biologically reasonable assessment of significance than would be obtained by permuting genes.</p>
<p>Finally, the third step consist on the adjustment for multiple hypothesis testing. When an entire database of gene sets is evaluated, the estimated significance level is adjusted for multiple hypothesis testing. First, the <span class="math inline">\(ES\)</span> is normalized for each gene set to account for the size of the set, yielding a normalized enrichment score (NES). Then, the proportion of false positives is controlled by calculating the FDR corresponding to each NES.</p>
<p>In 2010, a modification of the GSEA methodology was presented for metabolomics studies. This method was called Metabolite Set Enrichment Analysis (MSEA) and its main aim was to help researchers identify and interpret patterns of human and mammalian metabolite concentration changes in a biologically meaningful context <span class="citation">(Xia and Wishart <a href="#ref-xia2010msea" role="doc-biblioref">2010</a>)</span>. MSEA is currently widely used in the metabolomics community and it is implemented and freely available at the known <a href="https://www.metaboanalyst.ca">MetaboAnalyst</a> web-based tool <span class="citation">(Xia and Wishart <a href="#ref-xia2010msea" role="doc-biblioref">2010</a>)</span>.</p>
<p>As can be seen, GSEA approach is more complex than the ORA methodology, both in terms of methodological aspects and understanding of the method.</p>
<p>The <code>fobitools</code> package provides a function to perform MSEA using the FOBI information. This function requires a ranked list. Here, we will use the metabolites obtained in the limma model ranked by raw p-values.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">limma_FOBI_msea</span> <span class="op">&lt;-</span> <span class="va">limma_FOBI_names</span> <span class="op">%&gt;%</span>
  <span class="fu">select</span><span class="op">(</span><span class="va">FOBI</span>, <span class="va">P.Value</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">FOBI</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="op">-</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html">desc</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">P.Value</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="va">FOBI_msea</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html">as.vector</a></span><span class="op">(</span><span class="va">limma_FOBI_msea</span><span class="op">$</span><span class="va">P.Value</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">FOBI_msea</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">limma_FOBI_msea</span><span class="op">$</span><span class="va">FOBI</span>

<span class="va">msea_res</span> <span class="op">&lt;-</span> <span class="fu">fobitools</span><span class="fu">::</span><span class="fu"><a href="../reference/msea.html">msea</a></span><span class="op">(</span><span class="va">FOBI_msea</span>, pvalCutoff <span class="op">=</span> <span class="fl">0.06</span><span class="op">)</span>

<span class="va">msea_res</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kbl.html">kbl</a></span><span class="op">(</span>row.names <span class="op">=</span> <span class="cn">FALSE</span>, booktabs <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html">kable_styling</a></span><span class="op">(</span>latex_options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"striped"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<table class="table table">
<thead><tr>
<th style="text-align:left;">
className
</th>
<th style="text-align:right;">
classSize
</th>
<th style="text-align:right;">
log2err
</th>
<th style="text-align:right;">
ES
</th>
<th style="text-align:right;">
NES
</th>
<th style="text-align:right;">
pval
</th>
<th style="text-align:right;">
padj
</th>
<th style="text-align:left;">
leadingEdge
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
grape (whole, raw)
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.3807304
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
2.037782
</td>
<td style="text-align:right;">
0.0154703
</td>
<td style="text-align:right;">
0.2797203
</td>
<td style="text-align:left;">
FOBI:030590
</td>
</tr>
<tr>
<td style="text-align:left;">
grapefruit (whole, raw)
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0.1978220
</td>
<td style="text-align:right;">
0.9729730
</td>
<td style="text-align:right;">
1.982706
</td>
<td style="text-align:right;">
0.0509491
</td>
<td style="text-align:right;">
0.2797203
</td>
<td style="text-align:left;">
FOBI:030523
</td>
</tr>
<tr>
<td style="text-align:left;">
dairy food product
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.1882041
</td>
<td style="text-align:right;">
0.6383555
</td>
<td style="text-align:right;">
1.663786
</td>
<td style="text-align:right;">
0.0559441
</td>
<td style="text-align:right;">
0.2797203
</td>
<td style="text-align:left;">
FOBI:030701, FOBI:030704, FOBI:030697, FOBI:030692
</td>
</tr>
<tr>
<td style="text-align:left;">
egg food product
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.1882041
</td>
<td style="text-align:right;">
0.6383555
</td>
<td style="text-align:right;">
1.663786
</td>
<td style="text-align:right;">
0.0559441
</td>
<td style="text-align:right;">
0.2797203
</td>
<td style="text-align:left;">
FOBI:030701, FOBI:030704, FOBI:030697, FOBI:030692
</td>
</tr>
<tr>
<td style="text-align:left;">
meat food product
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.1882041
</td>
<td style="text-align:right;">
0.6383555
</td>
<td style="text-align:right;">
1.663786
</td>
<td style="text-align:right;">
0.0559441
</td>
<td style="text-align:right;">
0.2797203
</td>
<td style="text-align:left;">
FOBI:030701, FOBI:030704, FOBI:030697, FOBI:030692
</td>
</tr>
</tbody>
</table>
<p>As we can see, the enrichment analysis with the MSEA method seems to be much more accurate than the ORA method, since the two classes that head the results table (“<em>grape (whole, raw)</em>” and “<em>grapefruit (whole, raw)</em>”) are clearly within the FOBI food group (set) “<em>plant fruit food product</em>”, which is aligned with the study intervention, cranberry juice intake.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/fobi_graph.html">fobi_graph</a></span><span class="op">(</span>terms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"FOODON:03301123"</span>, <span class="st">"FOODON:03301702"</span><span class="op">)</span>, 
           get <span class="op">=</span> <span class="st">"anc"</span>, 
           labels <span class="op">=</span> <span class="cn">TRUE</span>, 
           labelsize <span class="op">=</span> <span class="fl">6</span><span class="op">)</span></code></pre></div>
<p><img src="MW_ST000291_enrichment_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
<div id="msea-plot-with-ggplot2" class="section level3">
<h3 class="hasAnchor">
<a href="#msea-plot-with-ggplot2" class="anchor"></a>MSEA plot with <code>ggplot2</code>
</h3>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggplot</span><span class="op">(</span><span class="va">msea_res</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">pval</span><span class="op">)</span>, y <span class="op">=</span> <span class="va">NES</span>, color <span class="op">=</span> <span class="va">NES</span>, size <span class="op">=</span> <span class="va">classSize</span>, label <span class="op">=</span> <span class="va">className</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">xlab</span><span class="op">(</span><span class="st">"-log10(P-value)"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ylab</span><span class="op">(</span><span class="st">"NES (Normalized Enrichment Score)"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggrepel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html">geom_label_repel</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"black"</span>, size <span class="op">=</span> <span class="fl">7</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">theme_bw</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"top"</span>,
        text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">22</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_color_viridis_c</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_size</span><span class="op">(</span>guide <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></code></pre></div>
<p><img src="MW_ST000291_enrichment_files/figure-html/unnamed-chunk-16-1.png" width="1152"></p>
</div>
<div id="network-of-metabolites-found-in-msea" class="section level3">
<h3 class="hasAnchor">
<a href="#network-of-metabolites-found-in-msea" class="anchor"></a>Network of metabolites found in MSEA</h3>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">FOBI_terms</span> <span class="op">&lt;-</span> <span class="va">msea_res</span> <span class="op">%&gt;%</span> 
  <span class="fu">unnest</span><span class="op">(</span>cols <span class="op">=</span> <span class="va">leadingEdge</span><span class="op">)</span>

<span class="fu">fobitools</span><span class="fu">::</span><span class="va"><a href="../reference/fobi.html">fobi</a></span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">FOBI</span> <span class="op">%in%</span> <span class="va">FOBI_terms</span><span class="op">$</span><span class="va">leadingEdge</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">pull</span><span class="op">(</span><span class="va">id_code</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="../reference/fobi_graph.html">fobi_graph</a></span><span class="op">(</span>get <span class="op">=</span> <span class="st">"anc"</span>, 
             labels <span class="op">=</span> <span class="cn">TRUE</span>, 
             legend <span class="op">=</span> <span class="cn">TRUE</span>, 
             labelsize <span class="op">=</span> <span class="fl">6</span>, 
             legendSize <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></code></pre></div>
<p><img src="MW_ST000291_enrichment_files/figure-html/unnamed-chunk-17-1.png" width="1152"></p>
</div>
</div>
</div>
<div id="limitations" class="section level1">
<h1 class="hasAnchor">
<a href="#limitations" class="anchor"></a>Limitations</h1>
<p>The FOBI ontology is currently in its first release version, so it does not yet include information on many metabolites and food relationships. All future efforts will be directed at expanding this ontology, leading to a significant increase in the number of metabolites and metabolite-food relationships. The <code>fobitools</code> package provides the methodology for easy use of the FOBI ontology regardless of the amount of information it contains. Therefore, future FOBI improvements will also have a direct impact on the <code>fobitools</code> package, increasing its utility and allowing to perform, among others, more accurate, complete and robust enrichment analyses.</p>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session Information</h1>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.0.2 (2020-06-22)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS  10.16</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] es_ES.UTF-8/es_ES.UTF-8/es_ES.UTF-8/C/es_ES.UTF-8/es_ES.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span>
<span class="co">#&gt; [8] methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] SummarizedExperiment_1.20.0  Biobase_2.50.0              </span>
<span class="co">#&gt;  [3] GenomicRanges_1.42.0         GenomeInfoDb_1.26.7         </span>
<span class="co">#&gt;  [5] IRanges_2.24.1               S4Vectors_0.28.1            </span>
<span class="co">#&gt;  [7] BiocGenerics_0.36.1          MatrixGenerics_1.2.1        </span>
<span class="co">#&gt;  [9] matrixStats_0.58.0           metabolomicsWorkbenchR_1.0.0</span>
<span class="co">#&gt; [11] POMA_1.1.13                  kableExtra_1.3.4            </span>
<span class="co">#&gt; [13] ggrepel_0.9.1                rvest_1.0.0                 </span>
<span class="co">#&gt; [15] forcats_0.5.1                stringr_1.4.0               </span>
<span class="co">#&gt; [17] dplyr_1.0.6                  purrr_0.3.4                 </span>
<span class="co">#&gt; [19] readr_1.4.0                  tidyr_1.1.3                 </span>
<span class="co">#&gt; [21] tibble_3.1.1                 ggplot2_3.3.3               </span>
<span class="co">#&gt; [23] tidyverse_1.3.1              fobitools_0.99.56           </span>
<span class="co">#&gt; [25] BiocStyle_2.18.1            </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] utf8_1.2.1                  tidyselect_1.1.1           </span>
<span class="co">#&gt;   [3] RSQLite_2.2.7               grid_4.0.2                 </span>
<span class="co">#&gt;   [5] BiocParallel_1.24.1         gmp_0.6-2                  </span>
<span class="co">#&gt;   [7] pROC_1.17.0.1               munsell_0.5.0              </span>
<span class="co">#&gt;   [9] codetools_0.2-18            ragg_1.1.2                 </span>
<span class="co">#&gt;  [11] preprocessCore_1.52.1       withr_2.4.2                </span>
<span class="co">#&gt;  [13] colorspace_2.0-1            highr_0.9                  </span>
<span class="co">#&gt;  [15] knitr_1.33                  rstudioapi_0.13            </span>
<span class="co">#&gt;  [17] mzID_1.28.0                 labeling_0.4.2             </span>
<span class="co">#&gt;  [19] GenomeInfoDbData_1.2.4      polyclip_1.10-0            </span>
<span class="co">#&gt;  [21] bit64_4.0.5                 farver_2.1.0               </span>
<span class="co">#&gt;  [23] rprojroot_2.0.2             vctrs_0.3.8                </span>
<span class="co">#&gt;  [25] generics_0.1.0              ipred_0.9-11               </span>
<span class="co">#&gt;  [27] xfun_0.22                   randomForest_4.6-14        </span>
<span class="co">#&gt;  [29] R6_2.5.0                    doParallel_1.0.16          </span>
<span class="co">#&gt;  [31] clue_0.3-59                 graphlayouts_0.7.1         </span>
<span class="co">#&gt;  [33] DelayedArray_0.16.3         bitops_1.0-7               </span>
<span class="co">#&gt;  [35] cachem_1.0.4                fgsea_1.16.0               </span>
<span class="co">#&gt;  [37] assertthat_0.2.1            scales_1.1.1               </span>
<span class="co">#&gt;  [39] vroom_1.4.0                 ggraph_2.0.5               </span>
<span class="co">#&gt;  [41] nnet_7.3-16                 gtable_0.3.0               </span>
<span class="co">#&gt;  [43] Cairo_1.5-12.2              affy_1.68.0                </span>
<span class="co">#&gt;  [45] tidygraph_1.2.0             timeDate_3043.102          </span>
<span class="co">#&gt;  [47] tictoc_1.0.1                rlang_0.4.11               </span>
<span class="co">#&gt;  [49] clisymbols_1.2.0            systemfonts_1.0.1          </span>
<span class="co">#&gt;  [51] mzR_2.24.1                  GlobalOptions_0.1.2        </span>
<span class="co">#&gt;  [53] splines_4.0.2               ModelMetrics_1.2.2.2       </span>
<span class="co">#&gt;  [55] impute_1.64.0               selectr_0.4-2              </span>
<span class="co">#&gt;  [57] broom_0.7.6                 RecordLinkage_0.4-12.1     </span>
<span class="co">#&gt;  [59] reshape2_1.4.4              BiocManager_1.30.12        </span>
<span class="co">#&gt;  [61] yaml_2.2.1                  modelr_0.1.8               </span>
<span class="co">#&gt;  [63] backports_1.2.1             caret_6.0-86               </span>
<span class="co">#&gt;  [65] tools_4.0.2                 lava_1.6.9                 </span>
<span class="co">#&gt;  [67] bookdown_0.22               affyio_1.60.0              </span>
<span class="co">#&gt;  [69] ellipsis_0.3.2              jquerylib_0.1.4            </span>
<span class="co">#&gt;  [71] ff_4.0.4                    RColorBrewer_1.1-2         </span>
<span class="co">#&gt;  [73] proxy_0.4-25                MSnbase_2.16.1             </span>
<span class="co">#&gt;  [75] MultiAssayExperiment_1.16.0 Rcpp_1.0.6                 </span>
<span class="co">#&gt;  [77] plyr_1.8.6                  zlibbioc_1.36.0            </span>
<span class="co">#&gt;  [79] RCurl_1.98-1.3              rpart_4.1-15               </span>
<span class="co">#&gt;  [81] GetoptLong_1.0.5            viridis_0.6.1              </span>
<span class="co">#&gt;  [83] haven_2.4.1                 cluster_2.1.2              </span>
<span class="co">#&gt;  [85] fs_1.5.0                    magrittr_2.0.1             </span>
<span class="co">#&gt;  [87] RSpectra_0.16-0             data.table_1.14.0          </span>
<span class="co">#&gt;  [89] circlize_0.4.12             reprex_2.0.0               </span>
<span class="co">#&gt;  [91] pcaMethods_1.82.0           ProtGenerics_1.22.0        </span>
<span class="co">#&gt;  [93] hms_1.0.0                   patchwork_1.1.1            </span>
<span class="co">#&gt;  [95] evaluate_0.14               xtable_1.8-4               </span>
<span class="co">#&gt;  [97] XML_3.99-0.6                readxl_1.3.1               </span>
<span class="co">#&gt;  [99] gridExtra_2.3               shape_1.4.5                </span>
<span class="co">#&gt; [101] compiler_4.0.2              ellipse_0.4.2              </span>
<span class="co">#&gt; [103] ncdf4_1.17                  crayon_1.4.1               </span>
<span class="co">#&gt; [105] htmltools_0.5.1.1           mgcv_1.8-35                </span>
<span class="co">#&gt; [107] corpcor_1.6.9               qdapRegex_0.7.2            </span>
<span class="co">#&gt; [109] lubridate_1.7.10            DBI_1.1.1                  </span>
<span class="co">#&gt; [111] tweenr_1.0.2                dbplyr_2.1.1               </span>
<span class="co">#&gt; [113] ComplexHeatmap_2.6.2        MASS_7.3-54                </span>
<span class="co">#&gt; [115] Matrix_1.3-2                permute_0.9-5              </span>
<span class="co">#&gt; [117] cli_2.5.0                   vsn_3.58.0                 </span>
<span class="co">#&gt; [119] gower_0.2.2                 textclean_0.9.3            </span>
<span class="co">#&gt; [121] evd_2.3-3                   RankProd_3.16.0            </span>
<span class="co">#&gt; [123] igraph_1.2.6                pkgconfig_2.0.3            </span>
<span class="co">#&gt; [125] pkgdown_1.6.1               recipes_0.1.16             </span>
<span class="co">#&gt; [127] MALDIquant_1.19.3           xml2_1.3.2                 </span>
<span class="co">#&gt; [129] foreach_1.5.1               rARPACK_0.11-0             </span>
<span class="co">#&gt; [131] svglite_2.0.0               ggcorrplot_0.1.3           </span>
<span class="co">#&gt; [133] bslib_0.2.4                 XVector_0.30.0             </span>
<span class="co">#&gt; [135] webshot_0.5.2               prodlim_2019.11.13         </span>
<span class="co">#&gt; [137] ada_2.0-5                   digest_0.6.27              </span>
<span class="co">#&gt; [139] vegan_2.5-7                 rmarkdown_2.7              </span>
<span class="co">#&gt; [141] cellranger_1.1.0            fastmatch_1.1-0            </span>
<span class="co">#&gt; [143] curl_4.3.1                  rjson_0.2.20               </span>
<span class="co">#&gt; [145] glasso_1.11                 nlme_3.1-152               </span>
<span class="co">#&gt; [147] lifecycle_1.0.0             jsonlite_1.7.2             </span>
<span class="co">#&gt; [149] mixOmics_6.14.1             desc_1.3.0                 </span>
<span class="co">#&gt; [151] viridisLite_0.4.0           limma_3.46.0               </span>
<span class="co">#&gt; [153] fansi_0.4.2                 pillar_1.6.0               </span>
<span class="co">#&gt; [155] ontologyIndex_2.7           lattice_0.20-44            </span>
<span class="co">#&gt; [157] fastmap_1.1.0               httr_1.4.2                 </span>
<span class="co">#&gt; [159] survival_3.2-11             glue_1.4.2                 </span>
<span class="co">#&gt; [161] png_0.1-7                   iterators_1.0.13           </span>
<span class="co">#&gt; [163] glmnet_4.1-1                bit_4.0.4                  </span>
<span class="co">#&gt; [165] ggforce_0.3.3               class_7.3-19               </span>
<span class="co">#&gt; [167] stringi_1.6.1               sass_0.3.1                 </span>
<span class="co">#&gt; [169] struct_1.2.0                blob_1.2.1                 </span>
<span class="co">#&gt; [171] textshaping_0.3.3           memoise_2.0.0              </span>
<span class="co">#&gt; [173] Rmpfr_0.8-4                 e1071_1.7-6</span></code></pre></div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-POMA">
<p>Castellano-Escuder, Pol, Cristina Andrés-Lacueva, and Alex Sánchez-Pla. 2021. <em>POMA: User-Friendly Workflow for Pre-Processing and Statistical Analysis of Mass Spectrometry Data</em>. <a href="https://github.com/pcastellanoescuder/POMA">https://github.com/pcastellanoescuder/POMA</a>.</p>
</div>
<div id="ref-castellano2020fobi">
<p>Castellano-Escuder, Pol, Raúl González-Domı́nguez, David S Wishart, Cristina Andrés-Lacueva, and Alex Sánchez-Pla. 2020. “FOBI: An Ontology to Represent Food Intake Data and Associate It with Metabolomic Data.” <em>Database</em> 2020.</p>
</div>
<div id="ref-creixell2015pathway">
<p>Creixell, Pau, Jüri Reimand, Syed Haider, Guanming Wu, Tatsuhiro Shibata, Miguel Vazquez, Ville Mustonen, et al. 2015. “Pathway and Network Analysis of Cancer Genomes.” <em>Nature Methods</em> 12 (7): 615.</p>
</div>
<div id="ref-limma">
<p>Ritchie, Matthew E, Belinda Phipson, Di Wu, Yifang Hu, Charity W Law, Wei Shi, and Gordon K Smyth. 2015. “limma Powers Differential Expression Analyses for RNA-Sequencing and Microarray Studies.” <em>Nucleic Acids Research</em> 43 (7): e47. <a href="https://doi.org/10.1093/nar/gkv007">https://doi.org/10.1093/nar/gkv007</a>.</p>
</div>
<div id="ref-subramanian2005gene">
<p>Subramanian, Aravind, Pablo Tamayo, Vamsi K Mootha, Sayan Mukherjee, Benjamin L Ebert, Michael A Gillette, Amanda Paulovich, et al. 2005. “Gene Set Enrichment Analysis: A Knowledge-Based Approach for Interpreting Genome-Wide Expression Profiles.” <em>Proceedings of the National Academy of Sciences</em> 102 (43): 15545–50.</p>
</div>
<div id="ref-sud2016metabolomics">
<p>Sud, Manish, Eoin Fahy, Dawn Cotter, Kenan Azam, Ilango Vadivelu, Charles Burant, Arthur Edison, et al. 2016. “Metabolomics Workbench: An International Repository for Metabolomics Data and Metadata, Metabolite Standards, Protocols, Tutorials and Training, and Analysis Tools.” <em>Nucleic Acids Research</em> 44 (D1): D463–D470.</p>
</div>
<div id="ref-xia2010msea">
<p>Xia, Jianguo, and David S Wishart. 2010. “MSEA: A Web-Based Tool to Identify Biologically Meaningful Patterns in Quantitative Metabolomic Data.” <em>Nucleic Acids Research</em> 38 (suppl_2): W71–W77.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Pol Castellano-Escuder, Cristina Andrés-Lacueva, Alex Sánchez-Pla.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
